central/plugins/org.jboss.tools.central/src/org/jboss/tools/central/installation/InstallationChecker.java

```diff
@@ -58,10 +60,26 @@ private InstallationChecker() {
 				JBossCentralActivator.log("Could not load default listing file for " + familyId);
 			}
 		}
+		
+		IProvisioningAgentProvider provider = (IProvisioningAgentProvider) PlatformUI.getWorkbench().getService(IProvisioningAgentProvider.class);
+		IProvisioningAgent agent;
+		agent = provider.createAgent(null); // null = location for running system
+		if (agent == null) {
+			throw new ProvisionException("Location was not provisioned by p2");
+		}
+		IProfileRegistry profileRegistry = (IProfileRegistry) agent.getService(IProfileRegistry.SERVICE_NAME);
+		if (profileRegistry == null) {
+			throw new ProvisionException("Unable to acquire the profile registry service.");
+		}
+		this.applicationProfile = profileRegistry.getProfile(IProfileRegistry.SELF);
+		if(this.applicationProfile == null) {
+			throw new ProvisionException("Current Eclipse instance doesn't support software installation.");
+		}
+		
 		this.installedUnitsPerFamily = new HashMap<String, Set<IInstallableUnit>>();
 	}
 	
-	public static InstallationChecker getInstance() {
+	public static InstallationChecker getInstance() throws ProvisionException {
```

Comment:
Then you should either use the Double Check Lock with a volatile instance (safe since Java 1.5), or make the method synchronized. Given this is not a highly concurrently accessed class, the synchronization overhead should be minimal